kubernetes/cache/cache_handlers.go

```diff
@@ -0,0 +1,48 @@
+package cache
+
+import (
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kiali/kiali/log"
+)
+
+type ServiceHandler struct {
+	Cache     KialiCache
+	Namespace string
+	cache.ResourceEventHandler
+}
+
+func NewServiceHandler(cache KialiCache, namespace string) ServiceHandler {
+	log.Infof("Adding a ServiceHandler for namespace [%s]", namespace)
+	return ServiceHandler{Cache: cache, Namespace: namespace}
+}
+
+func (sh ServiceHandler) OnAdd(obj interface{}) {
+	svc, ok := obj.(*core_v1.Service)
```

Comment:
Should only refresh the cache when Services are added/updated/removed? Perhaps ServiceEntries as well? And in general other resources that modify the registry?

Reply:
With Endpoints and ServiceEntries I think it could be good.
Also this cache is short term as it's using the tokenNamespace timeout, but I'm thinking that with this eventing mechanism it can have an increased period here.